### PR TITLE
Plugin facelift

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I have done some facelifting work during the investigation of [JENKINS-50216](https://issues.jenkins-ci.org/browse/JENKINS-50216).

- [x] Update to the newest Parent POM
- [x] Add Jenkinsfile to enable the PR builder

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.5</version>
+    <version>3.5</version>
```
Comment (jglick) [inline pom.xml:21]: 3.6 actually

File: pom.xml [hunk 2]
```diff
@@ -32,7 +32,16 @@
   <description>
     This plugin exposes a credential for use with the Git plugin for authenticating with Google source code hosting as a service account.
   </description>
-  <url>http://wiki.jenkins-ci.org/display/JENKINS/Google+Source+Plugin</url>
+  <url>https://wiki.jenkins.io/display/JENKINS/Google+Source+Plugin</url>
+
+  <properties>
+    <jenkins.version>1.625.3</jenkins.version>
+    <java.level>7</java.level>
+    <!-- TODO: JodaTime conflicts with the version coming from the optional Git dependency (1.5.1 vs. 2.4)-->
```
Comment (jglick) [inline pom.xml:40]: So use `<exclusions>`.

File: pom.xml [hunk 3]
```diff
@@ -187,7 +179,6 @@
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>4.11</version>
       <scope>test</scope>
     </dependency>
```
Comment (jglick) [inline pom.xml:183]: Just delete the dep—it is inherited anyway.

**Review by jglick — APPROVED** [review]

**Comment by gunan** [comment]
Adding @astroilov as my knowledge of the code accumulated quite some dust.

**Pull request merged**